base_layer/core/src/transactions/aggregated_body.rs

```diff
@@ -170,6 +170,66 @@ impl AggregateBody {
             .any(|input| self.outputs.iter().any(|o| o.is_equal_to(input)))
     }
 
+    /// This will check that there are no duplicated inputs. If the body is not sorted, this will sort the body
+    pub fn contains_duplicated_inputs(&self) -> bool {
+        // This can be expensive so we add in a sort check if its sorted
+        if self.sorted {
+            return self.contains_duplicated_inputs_sorted();
+        }
+        for i in 0..self.inputs().len() {
+            for j in (i + 1)..self.inputs().len() {
+                if self.inputs()[i] == self.inputs()[j] {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    /// This will check that there are no duplicated outputs. If the body is not sorted, this will sort the body
+    pub fn contains_duplicated_outputs(&self) -> bool {
+        // This can be expensive so we add in a sort check if its sorted
+        if self.sorted {
+            return self.contains_duplicated_outputs_sorted();
+        }
+        for i in 0..self.outputs().len() {
+            for j in (i + 1)..self.outputs().len() {
+                if self.outputs()[i] == self.outputs()[j] {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    /// This will check that there are no duplicated outputs, this will only work on a sorted transaction, will return
+    /// true if its not sorted. If the body is not sorted, this will sort the body
+    pub fn contains_duplicated_outputs_sorted(&self) -> bool {
+        if !self.sorted {
+            return true;
```

Comment:
This method should have been inlined and should definitely not be public. It's a bunch of extra code and now has a logic branch that makes no sense.

Reply:
Yeah, we should do something similar to what we do with the block body validation (i.e. transaction must be sorted on arrival, validate sorted and no dups in one loop)